fuels-core/src/code_gen/abigen.rs

```diff
@@ -111,18 +111,25 @@ impl Abigen {
         } else {
             (
                 quote! {
-                    use fuels_contract::contract::{Contract, ContractCall, CompiledContract};
+                    use fuels_contract::contract::{Contract, ContractCall};
                     use fuel_gql_client::client::FuelClient;
+                    use fuel_tx::ContractId;
+                    use std::str::FromStr;
                 },
                 quote! {
                     pub struct #name {
-                        compiled: CompiledContract,
+                        contract_id: ContractId,
                         fuel_client: FuelClient
                     }
 
                     impl #name {
-                        pub fn new(compiled: CompiledContract, fuel_client: FuelClient) -> Self {
-                            Self{ compiled, fuel_client }
+                        pub fn new(contract_id: String, fuel_client: FuelClient) -> Self {
+                            let contract_id = ContractId::from_str(&contract_id).unwrap();
+                            Self{ contract_id, fuel_client }
+                        }
+                        pub fn new_default(fuel_client: FuelClient) -> Self {
+                            let contract_id = ContractId::new([0xFFu8;32]);
```

Comment:
What is the reason for this function?

Reply:
I'm asking myself if there's a real good reason for it to exist. The only reason it exists is that in our test suite (`harness.rs`) we have some tests that instantiate a contract but don't make actual calls -- they're only testing stuff like the encoded data for a generated method call. So we're not deploying it. I'm guessing this `new_default` covers this need _but_ I guess we could get away by just having `new()` that takes the contract id, and in the tests that are not deploying, we just pass a zeroed contract id, like `new_default()` is doing; simplifying the whole API.

Reply:
Could `new_default` be non-`pub` if it's only to be used for testing? Or can the casts `impl` a `new_default` only in the test suite for this struct?

Reply:
The idea of this `new_default` is indeed to allow "cleaner" testing. Upon reading your comments, I realize it's not good to have this function exist in *every* contract. I pushed the alternative, which is to feed a *null* `contract_id` to the tests.